tests/Unit/DataStructures/Test_Tensor.cpp

```diff
@@ -502,6 +488,17 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.RankAndSize",
       symmetric_rank2_dim4{};
   CHECK(2 == symmetric_rank2_dim4.rank());
   CHECK(10 == symmetric_rank2_dim4.size());
+  double i = 1.948;  // not a value likely to be default-constructed to
+  for (auto& p : symmetric_rank2_dim4) {
+    p = i;
+    ++i;
+  }
+  CHECK(symmetric_rank2_dim4.get(0, 1) == symmetric_rank2_dim4.get(1, 0));
+  CHECK(symmetric_rank2_dim4.get(0, 2) == symmetric_rank2_dim4.get(2, 0));
+  CHECK(symmetric_rank2_dim4.get(0, 3) == symmetric_rank2_dim4.get(3, 0));
+  CHECK(symmetric_rank2_dim4.get(2, 1) == symmetric_rank2_dim4.get(1, 2));
+  CHECK(symmetric_rank2_dim4.get(3, 1) == symmetric_rank2_dim4.get(1, 3));
+  CHECK(symmetric_rank2_dim4.get(2, 3) == symmetric_rank2_dim4.get(3, 2));
```

Comment:
Add a check somewhere that some pair of entries that should be different are different.